### PR TITLE
🐛Pinecone destination: Fix loading of large records

### PR DESCRIPTION
**Pull request by flash1293**
This PR fixes two problems with loading large records with a lot of text:
* Pull in the changes from vector db CDK to chunk up strings to embed before sending them to the embedding API
* Do a similar fix in the connector itself - previously the connector would split up all chunks in the current batch into sets of 40 and send these in parallel (this is also recommended on the Pinecone website: https://docs.pinecone.io/docs/performance-tuning#using-the-grpc-client-to-get-higher-upsert-speeds ) The problem is that if there are too many parallel uploads, Pinecone will start rejecting them. This is solved on this PR by only parallelizing up to a specified limit

This can be tested in the following way:
* Get a huge record (like this one: https://gist.github.com/flash1293/697bf8238b0ab82532c2e7b986ff2db3 ) and put it into your examples folder
* Put a config object in `secrets/config.json`:
```
{
    "processing": {
        "text_fields": [
            "str_col", "title"
        ],
        "metadata_fields": [
            "int_col"
        ],
        "chunk_size": 1000
    },
    "embedding": {
        "mode": "openai",
        "openai_key": "mykey"
    },
    "indexing": {
        "pinecone_key": "mykey",
        "index": "myindex",
        "pinecone_environment": "my-env"
    }
}
```

Run from the command line via `cat examples/messages_huge.jsonl | python main.py write --config secrets/config.json --catalog examples/configured_catalog.json`. This will hit the openai rate limits a few times but will eventually succeed and upload all the chunks into the pinecone index.

### CODE REVIEW

File: airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py [hunk 1]
```diff
@@ -2,30 +2,22 @@
...
-    while chunk:
-        yield chunk
-        chunk = tuple(itertools.islice(it, batch_size))
-
-
 # large enough to speed up processing, small enough to not hit pinecone request limits
 PINECONE_BATCH_SIZE = 40
 
+# do not flood the server with too many connections in parallel
+PARALLELISM_LIMIT = 4
```
Comment (sherifnada) [inline airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py:19]: Fwiw this feels low? Was it empirically decided?

**Review by sherifnada — APPROVED** [review]
Nice!

File: airbyte-integrations/connectors/destination-pinecone/metadata.yaml [hunk 2]
```diff
@@ -9,10 +9,16 @@ data:
       enabled: true
     oss:
       enabled: true
+  resourceRequirements:
+    jobSpecific:
```
Comment (sherifnada) [inline airbyte-integrations/connectors/destination-pinecone/metadata.yaml:13]: you should leave a comment pointing to a ticket whose resolution would make this memory requirement unnecessary

**Review by sherifnada — APPROVED** [review]

**Comment by aaronsteers** [comment]
@flash1293 - Late to this PR, but I did a readthrough. No surprises, no issues from my side. 👍

**Pull request merged**